### ISSUE TITLE
Kotlin data classes not marking non-nullable fields as required

### ISSUE THREAD

**Issue by jonnii** [micronaut-projects/micronaut-openapi#550]
### Expected Behavior

Kotlin data classes with non nullable fields should have the swagger attributes marked as `required`

### Actual Behaviour

The fields are being marked as optionla

### Steps To Reproduce

https://github.com/jonnii/demo/commit/d30b76ba0a73ef669822cb9f507e990efd1c44b4

This commit shows an upgrade from `2.5.8` to `2.5.12` which exhibits this problem. 

### Example Application

https://github.com/jonnii/demo

### Version

2.5.12

**Comment by ilopmar** [comment]
We broke it during Micronaut 3 upgrade, this is the culprit https://github.com/micronaut-projects/micronaut-openapi/commit/2e21d4ef5509e6d8de54271ecc1891f45f36df4d. That commit is included in both `2.6.0` (shipped with Micronaut 2.5.12) and in `3.0.0`.

We released `3.0.0` because it contained the breaking change about mandatory params included in #540, but we didn't realize that it was a regression.

@jonnii If you force the version of the module to `3.0.0` then it should work for you. 
```groovy
configurations.all {
    resolutionStrategy {
        force 'io.micronaut.openapi:micronaut-openapi:3.0.0'
    }
}
```

@jameskleeh what do you think about included micronaut-openapi 3.0.0 in micronaut-core 2.5.x?

**Comment by ilopmar** [comment]
Fixed by #553

**Comment by ilopmar** [comment]
@jonnii The fix is included in micronaut-open 2.6.2. Please force that version manually until we release Micronaut 2.5.13 that will include it by default.

**Comment by jonnii** [comment]
great, thanks!
